src/chat.rs

```diff
@@ -4737,13 +4753,17 @@ pub(crate) async fn add_info_msg_with_cmd(
     timestamp_sent_rcvd: Option<i64>,
     parent: Option<&Message>,
     from_id: Option<ContactId>,
+    added_removed_id: Option<ContactId>,
 ) -> Result<MsgId> {
     let rfc724_mid = create_outgoing_rfc724_mid();
     let ephemeral_timer = chat_id.get_ephemeral_timer(context).await?;
 
     let mut param = Params::new();
     if cmd != SystemMessage::Unknown {
-        param.set_cmd(cmd)
+        param.set_cmd(cmd);
+    }
+    if let Some(contact_id) = added_removed_id {
+        param.set_int(Param::ContactAddedRemoved, contact_id.to_u32() as i32);
```

Comment:
`as` casts may panic, and since panics are super hard to debug, we try to avoid them where possible. While we could add another function for it, this is an easy solution: 
```suggestion
        param.set(Param::ContactAddedRemoved, contact_id.to_u32().to_string());
```